exercises/practice/two-fer/.approaches/introduction.md

```diff
@@ -0,0 +1,33 @@
+# Introduction
+
+## Default Paramater
+
+The key to this exercise is the optional paramater in the method definition.
+Using a default value there, rather than a conditional, makes the code much more readable, simple, and idiomatic.
+This is a pretty standard solution.
+
+```ruby
+module TwoFer
+  def self.two_fer(name = 'you')
+    "One for #{name}, one for me."
+  end
+end
+```
+
+## Method definition
+
+You can use `class` instead of `module` and use metaclasses instead of `self`, `module_function`, or extend `self`.
+
+## String method
+
+Solutions can also use string templates with `Kernel#sprintf`, `Kernel#format` or `String#%`
+
+```ruby
+module TwoFer
+  module_function
+​
+  def two_fer(name = 'you')
+    format('One for %<name>s, one for me.', name: name)
+  end
+end
+```
```

Comment:
Since it is mentioned that "You can use", we might add a singleton approach:

    ## Singleton
    
    You can use a constant, instead of `class` or `module` and define the
    method on this singleton object.
    
    ```ruby
    TwoFer = 'Sharing is fun.'
    
    def TwoFer.two_fer(name = 'you')
      'One for %<name>s, one for me.' % {name: name}
    end
    ```

Reply:
While there are a whole lot of things we _could_ do, I think that it's worth keeping the default "approaches" document focused on the more common solutions.

For more playful approaches I think we need to have more commentary around this, as it would otherwise be quite misleading as to what we would expect to see out in the wild.

Reply:
Then should we remove/change the word `can` and say instead that _the idiomatic approaches use things like_ because words like "can", "should", "must", "may" are powerful words.

How do we avoid discouraging the playful, where it is often where deeper understanding stems from?

Reply:
Or perhaps we that while we can do the listed things, it is not a complete list?  That is probably the thing that is missing here.

Reply:
These are good questions, @kotp. I believe the intention here is that we can do the listed things but that it's not a complete list.

@ErikSchierboom Should we tweak the wording to make this a bit clearer?

Reply:
I'd be fine with tweaking the wording a bit.
I've added some wording suggestions.

Reply:
I think the bit about this being an idiomatic approach solves it for me.

Reply:
I think so too.